src/Integration.Vsix/CFamily/CFamilyHelper.cs

```diff
@@ -44,8 +44,28 @@ internal static partial class CFamilyHelper
         private static readonly string analyzerExeFilePath = Path.Combine(
             CFamilyFilesDirectory, "subprocess.exe");
 
+        internal class PCHRequestLogger : ILogger
+        {
+            public void WriteLine(string message)
+            {
+                // We don't log in PCH request to avoid noise while navigating files
+            }
+
+            public void WriteLine(string messageFormat, params object[] args)
+            {
+                // We don't log in PCH request to avoid noise while navigating files
+            }
+        }
+
+        private static readonly PCHRequestLogger pchLogger = new PCHRequestLogger();
+
         public static Request CreateRequest(ILogger logger, ProjectItem projectItem, string absoluteFilePath, ICFamilyRulesConfigProvider cFamilyRulesConfigProvider, IAnalyzerOptions analyzerOptions)
         {
+            if (analyzerOptions is CFamilyAnalyzerOptions cFamilyAnalyzerOptions && cFamilyAnalyzerOptions.CreatePreCompiledHeaders)
```

Comment:
@abbas-sabra-sonarsource this only disables the new "not supported" notifications, right?
If so, I'm not sure if it really matters that the users sees them or not. The user is gonna see that we don't support it, and then just stop looking at the SonarLint pane, no? WDYT?

Reply:
Yes, it is to disable not supported notifications. In most cases, there will be files that we don't support in the same projects. This usually makes my SonarLint pane Full of irrelevant information of files that I'm just navigating. I know they are not supported, I'm not triggering any analysis they still fill my pane

Reply:
Ok, I understand.
I think this issue should be handled differently. Overriding the logger feels wrong: it would block us from logging meaningful messages, such as real failures (this PR now actually hides one of them -- the try-catch inside TryGetConfig). 
I think the approach should be different, such as perhaps creating log levels and setting a different verbosity for PCH, instead of disabling it completely (https://github.com/SonarSource/sonarlint-visualstudio/issues/788). It's a much bigger change but I think if we want to control the output, that should be the way to go.
@duncanp-sonar what do you think?

Reply:
`would block us from logging meaningful messages`
True, but **only while navigating** which sounded fine functionally. Changing the verbosity in the case of PCH would solve the problem too

Reply:
@abbas-sabra-sonarsource  I'm confused; by "navigating" do you mean "opening", in which case won't we still show the "not supported" when we run the analysis on file open anyway?

Reply:
FYI I've created ticket #1770 for this issue (we have an informal policy that any code changes should relate to a ticket). @abbas-sabra-sonarsource could you paste an example of verbose output in the ticket please? Thanks.

Reply:
@duncanp-sonar by navigating I don't mean opening, I mean focusing on a file or jumping between already opened files.
Showing the "not supported" on a file open is fine. It is actually useful. The problem is having "file not supported" for the same file as many times as you focus on it

Reply:
Thanks @abbas-sabra-sonarsource, makes sense. If it's just the "unsupported type" message we want to avoid for PCH requests then I think it would be better to conditionally suppress the logging in `TryGetConfig` for PCH requests, rather than disabling PCH logging completely.

Reply:
@duncanp-sonar,  I'm not disabling logging for PCH completely, I'm only disabling it in fetching the Request. Yes, the only concern is "unsupported type". Currently, all the logging done in CreateRequest is related to unsupported types. In principle, I think if CreateRequest is logging something, it must be about "request failure" => "unsupported type" => you don't want to see the message each time you focus on the file. 
logging through a function that checks if it is a PCH request before logging would work, but I thought that the overriding approach is nicer since it is future proof. I expect that we always want to hide logs that will be added later.

For example, the log mentioned by @rita-g-sonarsource : `Unable to collect C/C++ configuration for..`
This log implies that the file type isn't supported and we want to see it only when we analyze the file **instead of** every time we focus on it.

Let me know what do you think and if you still prefer the conditional suppressing.

Reply:
Ok, makes sense. In that case, I don't have a preference between the two approaches, although if you use the no-op logger approach I think it should be renamed. `PCHRequestLogger` suggests that it is used for the whole PCH request, not just when collecting the configuration. Maybe something like "NoOpLogger" instead, with an appropriate comment at line 66 in place of the two comments in line 51 and 56.

Reply:
Great! I'll update the PR